vault/identity_store.go

```diff
@@ -161,8 +174,14 @@ func (i *IdentityStore) Invalidate(ctx context.Context, key string) {
 		// storage entry is non-nil, its an indication of an update. In this
 		// case, entities in the updated bucket needs to be reinserted into
 		// MemDB.
+		parsedCount := 0
 		if bucket != nil {
-			for _, item := range bucket.Items {
+			items := make([]*storagepacker.Item, 0, len(bucket.Items)+len(bucket.ItemMap))
+			items = append(items, bucket.Items...)
+			for id, message := range bucket.ItemMap {
```

Comment:
Would there be a performance improvement if we do a range on `bucket.Items` and `bucket.ItemMap` separately instead of populating the `items` first and then ranging over it?

Reply:
Yeah that's fair. I think in reality only one of these will be non-zero at any given time so it's mostly defense in depth.